### ISSUE TITLE
CD pipeline doesn't pick up Flux changes!

### ISSUE THREAD

**Issue by CGHoussem** [Azure/arc-cicd-demo-src#21]
Hello,

I was following the guide on implementing CI/CD with GitOps (Flux v2).

On the [**CD pipeline section**](https://docs.microsoft.com/en-us/azure/azure-arc/kubernetes/tutorial-gitops-flux2-ci-cd#cd-pipeline) of the guide, **step 5** (_Monitor the Git Commit status on the Commit history tab. Once it is `succeeded` the CD pipeline will go ahead and start automated testing_), I always get "**failed**" as the Git Commit Status on the branch "**arc-cicd-demo-src**".

The pipeline fails at the task of "**Waiting for changes to be applied**" (it runs until I get a timeout. approx after an hour).

![image](https://user-images.githubusercontent.com/18559408/172342351-03169203-27e0-4bb7-b916-af89d2b5f5a6.png)

Please note that the Flux picked up the changes and successfully applied them in 10 minutes.

![image](https://user-images.githubusercontent.com/18559408/172339622-6448e414-ec74-4c48-946a-71da974b5ec8.png)

**Comment by eedorenko** [comment]
Check that the `Azure Repos PAT token` (the one you configured in --set orchestratorPAT=<Azure Repos PAT token>) should have `Build: Read & execute` and `Code: Read` permissions.

If it doesn't help, give me logs of GitOps Connector for the timeframe when Flux applied the changes.

**Comment by CGHoussem** [comment]
I found that there was a typo on the `Azure Repos PAT token`. I got it fixed and then redeployed the new version. (I've deleted the previous pod of course)

I've tried deploying the project again and I got the same result. I monitored the log of **the gitops-connector pod** via  `kubectl logs -l=app=gitops-connector -n flux-system -f`. Here is a preview of the log:

```
INFO:root:Starting periodic PR cleanup
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): dev.azure.com:443
DEBUG:urllib3.connectionpool:https://dev.azure.com:443 "GET /REDACTED/TestProject/_apis/git/repositories/arc-cicd-demo-gitops/pullRequests?searchCriteria.status=abandoned&api-version=6.0 HTTP/1.1" 200 None
INFO:root:Finished PR cleanup, sleeping for 30 seconds...
INFO:root:Starting periodic PR cleanup
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): dev.azure.com:443
DEBUG:urllib3.connectionpool:https://dev.azure.com:443 "GET /REDACTED/TestProject/_apis/git/repositories/arc-cicd-demo-gitops/pullRequests?searchCriteria.status=abandoned&api-version=6.0 HTTP/1.1" 200 None
INFO:root:Finished PR cleanup, sleeping for 30 seconds...
INFO:root:Starting periodic PR cleanup
DEBUG:urllib3.connectionpool:Starting new HTTPS connection (1): dev.azure.com:443
DEBUG:urllib3.connectionpool:https://dev.azure.com:443 "GET /REDACTED/TestProject/_apis/git/repositories/arc-cicd-demo-gitops/pullRequests?searchCriteria.status=abandoned&api-version=6.0 HTTP/1.1" 200 None
INFO:root:Finished PR cleanup, sleeping for 30 seconds...
```

Note that the `Azure Repos PAT token` has full permissions.

**Comment by nilakky** [comment]
Is this fixed?
We have noticed that if we approve PR manually then its working.

**Comment by CGHoussem** [comment]
> Is this fixed? We have noticed that if we approve PR manually then its working.

Unfortunately not, it hasn't been fixed. I haven't got a reply and now I lost access to the project.

**Comment by nilakky** [comment]
We have found issue with commit message. If there any special characters in merge message then its not picking by gitops connector

**Comment by atkei** [comment]
I think this tutorial does not work with Flux v2 because the current gitops-connector v1.2.0 does not include the following.
https://github.com/Azure/gitops-connector/pull/49

**Comment by eedorenko** [comment]
Fixed in v1.2.1
